sycl/test/separate-compile/cpp_version_mismatch_test_2.cpp

```diff
@@ -0,0 +1,63 @@
+// >> ---- device compilation
+// RUN: %clangxx -std=c++14 -fsycl-device-only -Xclang -fsycl-int-header=sycl_ihdr_a.h %s -c -o a_kernel.bc -I %sycl_include
+//
+// >> ---- host compilation: no cpp version mismatch
+// RUN: %clangxx -std=c++14 -include sycl_ihdr_a.h -c %s -I %sycl_include
+//
+// >> ---- code generation checks
+// RUN: FileCheck -input-file=sycl_ihdr_a.h %s
+// CHECK: #define STD_CPP_VERSION 
+// CHECK-NEXT: #if __cplusplus != STD_CPP_VERSION
+// CHECK-NEXT: #error "C++ version for host compilation does not match C++ version used for device compilation"
+// CHECK-NEXT: #endif

```

Comment:
Please do integration header testing in FE tests. Such tests are placed in clang/test/CodeGenSYCL .

Reply:
Capturing llvm-lit test transfer into a separate issue.

Reply:
What is the problem with llvm-lit?